BoxContentSDK/BoxContentSDK/BOXURLRequestSerialization.m

```diff
@@ -1210,13 +1208,6 @@ - (NSInteger)readData:(NSData *)data
 }
 
 - (BOOL)transitionToNextPhase {
-    if (![[NSThread currentThread] isMainThread]) {
-        dispatch_sync(dispatch_get_main_queue(), ^{
-            [self transitionToNextPhase];
-        });
-        return YES;
-    }
-
```

Comment:
Can this ever be called in a context where there is not a current runloop, or the runloop is ephemeral? It seems that that might be the reason it's dispatching to the main thread, but maybe it's not a concern.

Reply:
the way we're using it in content sdk, we won't actually use runloop. The inputstream we pass in is BOXMultipartBodyStream which overrides scheduleInRunLoop method to be no-op.